lib/data/external/AzureClient.js

```diff
@@ -206,6 +220,42 @@ class AzureClient {
             return callback();
         });
     }
+
+    copyObject(request, sourceKey, sourceLocationConstraintName, log,
+    callback) {
+        const destContainerName = request.bucketName;
+        const destObjectKey = request.objectKey;
+
+        const destAzureKey = this._createAzureKey(destContainerName,
+          destObjectKey, this._bucketMatch);
+
+        const sourceContainerName =
+        config.locationConstraints[sourceLocationConstraintName]
+        .details.azureContainerName;
+
+        this._client.startCopyBlob(`${this._azureBlobEndpoint}/` +
+          `${sourceContainerName}/${sourceKey}?${this._azureBlobSAS}`,
+        this._azureContainerName, destAzureKey, err => {
+            if (err) {
+                if (err.code === 'CannotVerifyCopySource') {
+                    logHelper(log, 'error', 'Unable to access ' +
+                    `${sourceContainerName} Azure Container`, err,
+                    this._dataStoreName);
+                    return callback(errors.AccessDenied
+                      .customizeDescription('Error: Unable to access ' +
+                      `${sourceContainerName} Azure Container`)
+                    );
+                }
+                logHelper(log, 'error', 'error from data backend on ' +
+                'copyObject', err, this._dataStoreName);
+                return callback(errors.InternalError
+                  .customizeDescription('Error returned from ' +
+                  `AWS: ${err.message}`)
+                );
+            }
```

Comment:
we need to check the response from azure to see if the copy was successful or "pending".  If it is pending we should return an error such as InvalidObjectState with a customized description that the azure status is pending.  They don't guarantee that the call is done.  See -- https://docs.microsoft.com/en-us/rest/api/storageservices/copy-blob

Reply:
ok thanks

Reply:
@LaurenSpiegel I aborted it when status was pending.

Reply:
As discussed, we will only do Azure server side copy if it is between the same account (which we know based on the endpoint). Otherwise, we will do our regular copy mechanism of get and put.